signer/src/stacks/api.rs

```diff
@@ -1275,45 +1296,113 @@ where
         .map_err(serde::de::Error::custom)
 }
 
+/// Extract a set of public keys from a Clarity value.
+///
+/// The value is expected to be a sequence of 33 byte buffers, each of
+/// which represents a compressed public key.
+fn extract_signer_set(value: Value) -> Result<BTreeSet<PublicKey>, Error> {
+    match value {
+        // Iterate through each record in the list and convert it to a
+        // public key. If the record is not a buffer, then return an error.
+        Value::Sequence(SequenceData::List(ListData { data, .. })) => data
+            .into_iter()
+            .flat_map(|item| extract_public_key(item).transpose())
+            .collect(),
+        // We expected the top-level value to be a list of buffers,
+        // but we got something else.
+        _ => Err(Error::InvalidStacksResponse(
+            "expected a sequence but got something else",
+        )),
+    }
+}
+
+/// Extract a public key from a Clarity value.
+///
+/// In the sbtc-registry smart contract, the aggregate key is stored in the
+/// `current-aggregate-pubkey` data var and is initialized to the 0x00
+/// byte, allowing use to distinguish between the initial value and an
+/// actual public key in that case. Ok(None) is returned if the value is
+/// the initial value.
+fn extract_public_key(value: Value) -> Result<Option<PublicKey>, Error> {
+    match value {
+        Value::Sequence(SequenceData::Buffer(BuffData { data })) => {
+            // The initial value of the data var is all zeros
+            if data.iter().all(|v| *v == 0) {
+                Ok(None)
+            } else {
+                PublicKey::from_slice(&data).map(Some)
+            }
+        }
+        _ => Err(Error::InvalidStacksResponse(
+            "expected a buffer but got something else",
+        )),
+    }
+}
+
+/// Extract a signature threshold from a Clarity value.
+///
+/// In the sbtc-registry smart contract, the signature threshold is stored
+/// in the `current-signature-threshold` data var and is initialized to 0,
+/// allowing use to distinguish between the initial value and an actual
+/// signature threshold. Ok(None) is returned if the value is the initial
+/// value.
+fn extract_signatures_required(value: Value) -> Result<Option<u16>, Error> {
+    match value {
+        Value::UInt(0) => Ok(None),
```

Comment:
Do we really want represent 0 signatures required as `None` and do not just use 0 ?

Reply:
Yeah I think we want it to be `None`. My reasoning is that the signer's behavior depends on whether the registry has signer set data or not. Since the signers cannot set the `signatures-required` to zero, so seeing zero here means that it's unset and that the registry does not have signer set data. Therefore, returning `None` makes sense. Does that make sense?